### PR TITLE
fix: use GitHub App token when authed with GitHub App

### LINKED ISSUE #98: Project Support Permissions
### Describe the bug

When using the `PROJECT_ID` environment variable, getting the following error:

```
Request failed: 401 Client Error: Unauthorized for url: https://api.github.com/graphql
```

I am authenticating using a GitHub app with the following permissions:

- Read access to metadata
- Read and write access to administration, code, issues, pull requests, repository custom properties, repository projects, and security events
- Admin access to organization projects

### To Reproduce

1. Setup github action
2. Use GitHub app to authenticate
3. Use the permissions described above
4. Create a project
5. Run with batch size of 1

### Expected behavior

The PR that is created should be associated to the project of the id that is passed in.

### Screenshots

_No response_

### Things that I have tried:

- Every permission that _could_ make sense
- I have triple-checked that the installation has been installed properly to the organization
- Debug mode (didn't get any extra output, unfortunately)

### Environment Variables Used

```
GH_APP_ID: "<redacted>"
GH_APP_INSTALLATION_ID: "<redacted>"
GH_APP_PRIVATE_KEY: ${{ secrets.GH_APP_PRIVATE_KEY }}
ORGANIZATION: <redacted>
COMMIT_MESSAGE: "chore: add dependabot.yml"
TITLE: "Add dependabot configuration"
BODY: "Please add this dependabot configuration so that we can keep the dependencies in this repo up to date and secure. For more information on dependabot configuration, check out the [official docs](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file)."
GROUP_DEPENDENCIES: true
PROJECT_ID: 9
BATCH_SIZE: 50
```

### PR DESCRIPTION
**Pull request by jmeridth**
Fixes #98

# Pull Request

## Proposed Changes

Currently we are still trying to use the GH_TOKEN env var when making graphql calls even after authenticating with a GitHub App Installation. This change fixes that.

Also fixed a few other things while in here:

- [x] add all missing env var options to .env-example
- [x] ensure .env file contents are not used in env var tests
- [x] alphabetize things, as they should be
- [x] add test for github auth change token usage

### Author/Contributor

- [x] If documentation is needed for this change, has that been included in this pull request
- [x] run `make lint` and fix any issues that you have introduced
- [x] run `make test` and ensure you have test coverage for the lines you are introducing

### Reviewer

- [x] Label as either `bug`, `documentation`, `enhancement`, `infrastructure`, `maintenance` or `breaking`

### CODE REVIEW

**Comment by jmeridth** [comment]
Still testing this

File: auth.py [hunk 1]
```diff
@@ -41,3 +42,29 @@ def auth_to_github(
...
+    Args:
+        gh_app_id (str): the GitHub App ID
+        gh_app_private_key_bytes (bytes): the GitHub App Private Key
+        gh_app_installation_id (str): the GitHub App Installation ID
+
+    Returns:
+        str: the GitHub App token
+    """
+    jwt_headers = github3.apps.create_jwt_headers(gh_app_private_key_bytes, gh_app_id)
+    url = f"https://api.github.com/app/installations/{gh_app_installation_id}/access_tokens"
```
Comment (jmeridth) [inline auth.py:62]: core change of this PR

File: .env-example [hunk 2]
```diff
@@ -1,7 +1,21 @@
...
+EXEMPT_ECOSYSTEMS=""
+EXEMPT_REOPS=""
+FILTER_VISIBILITY=""
+GH_TOKEN=""
+GROUP_DEPENDENCIES=""
+ORGANIZATION=""
+PROJECT_ID=""
+REPOSITORY=""
+TITLE=""
+TYPE=""
```
Comment (zkoppert) [inline .env-example:16]:
```suggestion
BATCH_SIZE = ""
BODY = ""
COMMIT_MESSAGE = ""
CREATED_AFTER_DATE = ""
DRY_RUN = ""
ENABLE_SECURITY_UPDATES = ""
EXEMPT_ECOSYSTEMS = ""
EXEMPT_REOPS = ""
FILTER_VISIBILITY = ""
GH_TOKEN = ""
GROUP_DEPENDENCIES = ""
ORGANIZATION = ""
PROJECT_ID = ""
REPOSITORY = ""
TITLE = ""
TYPE = ""
```

File: .env-example [hunk 3]
```diff
@@ -1,7 +1,21 @@
...
 
 # GITHUB APP
-GH_APP_ID = ""
-GH_INSTALLATION_ID = ""
-GH_PRIVATE_KEY = ""
+GH_APP_ID=""
+GH_INSTALLATION_ID=""
+GH_PRIVATE_KEY=""
```
Comment (zkoppert) [inline .env-example:21]:
```suggestion
GH_APP_ID = ""
GH_INSTALLATION_ID = ""
GH_PRIVATE_KEY = ""
```

File: env.py [hunk 4]
```diff
@@ -95,7 +93,7 @@ def get_env_vars(
         exempt_ecosystems_list (list[str]): A list of package ecosystems to exempt from the action
     """
     if not test:
-        # Load from .env file if it exists
+        # Load from .env file if it exists and not tests
```
Comment (zkoppert) [inline env.py:96]:
```suggestion
        # Load from .env file if it exists and not testing
```

File: auth.py [hunk 5]
```diff
@@ -41,3 +42,29 @@ def auth_to_github(
...
+
+    try:
+        response = requests.post(url, headers=jwt_headers, json=None, timeout=20)
```
Comment (jmeridth) [inline auth.py:65]:
```suggestion
        response = requests.post(url, headers=jwt_headers, json=None, timeout=5)
```

**Review by zkoppert — APPROVED** [review]

**Pull request merged**